// redux-form-website-template/es/components/GithubButton.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var GithubButton = function GithubButton(_ref) {
  var user = _ref.user,
      repo = _ref.repo,
      type = _ref.type,
      width = _ref.width,
      height = _ref.height,
      count = _ref.count,
      large = _ref.large;

  var src = 'https://ghbtns.com/github-btn.html?user=' + user + '&repo=' + repo + '&type=' + type;
  if (count) {
    src += '&count=true';
  }
  if (large) {
    src += '&size=large';
  }
  return _react2.default.createElement('iframe', {
    src: src,
    title: 'GitHub Buttons',
    frameBorder: '0',
    allowTransparency: 'true',
    scrolling: '0',
    width: width,
    height: height,
    style: { border: 'none', width: width, height: height }
  });
};

GithubButton.propTypes = {
  user: _propTypes2.default.string.isRequired,
  repo: _propTypes2.default.string.isRequired,
  type: _propTypes2.default.oneOf(['star', 'watch', 'fork', 'follow']).isRequired,
  width: _propTypes2.default.number.isRequired,
  height: _propTypes2.default.number.isRequired,
  count: _propTypes2.default.bool,
  large: _propTypes2.default.bool
};

exports.default = GithubButton;
