// redux-form-website-template/es/components/Nav.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require("classnames");

var _classnames2 = _interopRequireDefault(_classnames);

var _marked = require("marked");

var _marked2 = _interopRequireDefault(_marked);

var _Sponsor = require("./Sponsor");

var _Sponsor2 = _interopRequireDefault(_Sponsor);

var _Nav = require("./Nav.scss");

var _Nav2 = _interopRequireDefault(_Nav);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var formatLabel = function formatLabel(label) {
  return (/<p>(.+)<\/p>/.exec((0, _marked2.default)(label))[1]
  );
};

var Nav = function (_Component) {
  _inherits(Nav, _Component);

  function Nav(props) {
    _classCallCheck(this, Nav);

    var _this = _possibleConstructorReturn(this, (Nav.__proto__ || Object.getPrototypeOf(Nav)).call(this, props));

    _this.open = _this.open.bind(_this);
    _this.close = _this.close.bind(_this);
    _this.state = {
      open: false
    };
    return _this;
  }

  _createClass(Nav, [{
    key: "renderItem",
    value: function renderItem(href, label) {
      var indent = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : 0;
      var _props = this.props,
          path = _props.path,
          url = _props.url;

      return _react2.default.createElement("a", {
        href: "" + (url || "") + href,
        className: (0, _classnames2.default)(_Nav2.default["indent" + indent], _defineProperty({}, _Nav2.default.active, href === path)),
        dangerouslySetInnerHTML: { __html: formatLabel(label) }
      });
    }
  }, {
    key: "open",
    value: function open() {
      this.setState({ open: true });
    }
  }, {
    key: "close",
    value: function close() {
      this.setState({ open: false });
    }
  }, {
    key: "render",
    value: function render() {
      var open = this.state.open;
      var url = this.props.url;

      return _react2.default.createElement(
        "div",
        { className: (0, _classnames2.default)(_Nav2.default.nav, _defineProperty({}, _Nav2.default.open, open)) },
        _react2.default.createElement("button", { type: "button", onClick: this.open }),
        _react2.default.createElement(
          "div",
          { className: _Nav2.default.overlay, onClick: this.close },
          _react2.default.createElement("i", { className: "fa fa-times" }),
          " Close"
        ),
        _react2.default.createElement("div", { className: _Nav2.default.placeholder }),
        _react2.default.createElement(
          "nav",
          { className: _Nav2.default.menu },
          _react2.default.createElement(
            "a",
            { href: url, className: _Nav2.default.brand },
            "Redux Form"
          ),
          _react2.default.createElement(_Sponsor2.default, null),
          this.renderItem("/docs/GettingStarted.md", "Getting Started"),
          this.renderItem("/docs/MigrationGuide.md", "`v6` Migration Guide"),
          this.renderItem("/docs/ValueLifecycle.md", "Value Lifecycle"),
          this.renderItem("/docs/Flow.md", "Flow"),
          this.renderItem("/docs/api", "API"),
          this.renderItem("/docs/api/ReduxForm.md", "`reduxForm()`", 1),
          this.renderItem("/docs/api/Props.md", "`props`", 1),
          this.renderItem("/docs/api/Field.md", "`Field`", 1),
          this.renderItem("/docs/api/Fields.md", "`Fields`", 1),
          this.renderItem("/docs/api/FieldArray.md", "`FieldArray`", 1),
          this.renderItem("/docs/api/Form.md", "`Form`", 1),
          this.renderItem("/docs/api/FormName.md", "`FormName`", 1),
          this.renderItem("/docs/api/FormSection.md", "`FormSection`", 1),
          this.renderItem("/docs/api/FormValues.md", "`formValues()`", 1),
          this.renderItem("/docs/api/FormValueSelector.md", "`formValueSelector()`", 1),
          this.renderItem("/docs/api/Reducer.md", "`reducer`", 1),
          this.renderItem("/docs/api/ReducerPlugin.md", "`reducer.plugin()`", 2),
          this.renderItem("/docs/api/SubmissionError.md", "`SubmissionError`", 1),
          this.renderItem("/docs/api/ActionCreators.md", "Action Creators", 1),
          this.renderItem("/docs/api/Selectors.md", "Selectors", 1),
          this.renderItem("/docs/faq", "FAQ"),
          this.renderItem("/examples", "Examples"),
          this.renderItem("/examples/simple", "Simple Form", 1),
          this.renderItem("/examples/syncValidation", "Sync Validation", 1),
          this.renderItem("/examples/fieldLevelValidation", "Field-Level Validation", 1),
          this.renderItem("/examples/submitValidation", "Submit Validation", 1),
          this.renderItem("/examples/asyncValidation", "Async Blur Validation", 1),
          this.renderItem("/examples/asyncChangeValidation", "Async Change Validation", 1),
          this.renderItem("/examples/initializeFromState", "Initializing from State", 1),
          this.renderItem("/examples/selectingFormValues", "Selecting Form Values", 1),
          this.renderItem("/examples/fieldArrays", "Field Arrays", 1),
          this.renderItem("/examples/remoteSubmit", "Remote Submit", 1),
          this.renderItem("/examples/normalizing", "Normalizing", 1),
          this.renderItem("/examples/immutable", "Immutable JS", 1),
          this.renderItem("/examples/wizard", "Wizard Form", 1),
          this.renderItem("/examples/material-ui", "Material UI", 1),
          this.renderItem("/examples/react-widgets", "React Widgets", 1),
          this.renderItem("/docs/DocumentationVersions.md", "Older Versions")
        )
      );
    }
  }]);

  return Nav;
}(_react.Component);

Nav.propTypes = {
  path: _propTypes2.default.string.isRequired,
  url: _propTypes2.default.string.isRequired
};

exports.default = Nav;
